pkg/server/manager/pipeline.go

```diff
@@ -386,16 +385,31 @@ func (m *pipelineManager) UpdatePipeline(projectName string, pipelineName string
 	if pipeline.AutoTrigger != nil && pipeline.AutoTrigger.SCMTrigger != nil {
 		scmTrigger := pipeline.AutoTrigger.SCMTrigger
 		if scmTrigger.Webhook != "" {
-			if err := provider.DeleteWebHook(gitSource.Url, scmTrigger.Webhook); err != nil {
+			if err := provider.DeleteWebHook(oldGitSource.Url, scmTrigger.Webhook); err != nil {
 				return nil, err
 			}
 		}
 	}
 
+	var newGitSource *api.GitSource
```

Comment:
Create webhook for new repo before delete old webhook, then can directly return if there is error.
This can avoid the risk of failing to recreate old webhook later.

Reply:
If new URL equals the old one, new webhook will be deleted.